mlir/include/mlir/Dialect/Arith/IR/ArithBase.td

```diff
@@ -133,4 +133,27 @@ def Arith_FastMathAttr :
   let assemblyFormat = "`<` $value `>`";
 }
 
+//===----------------------------------------------------------------------===//
+// IntegerOverflowFlags
```

Comment:
This seems to break the Python bindings, since now both the arith and llvm dialects define `IntegerOverflowFlags`:

```
Traceback (most recent call last):
  ...
File "[...]/mlir/ir.py", line 14, in decorator_builder
    AttrBuilder.insert(kind, func, replace=replace)
RuntimeError: Attribute builder for 'IntegerOverflowFlags' is already registered with func: <function _integeroverflowflags at 0x7faf26f8ce00>
```

The repro is

```
from mlir.dialects import arith, llvm
```

Reply:
I was wondering why this wasn't the case for fastmath flags and they are actually named slightly differently between llvm and arith (`FastMathFlags` vs `FastmathFlags`).

Anyways, I'm not familiar with in-tree MLIR python bindings and considering these enums live in different namespaces on C++ level, this sounds like quite a big limitation, which people will continue to hit randomly.

Any ideas how to fix it properly?

Reply:
Just need to namespace (by dialect) the generated enum bindings [here](https://github.com/llvm/llvm-project/blob/main/mlir/tools/mlir-tblgen/EnumPythonBindingGen.cpp#L108) and [here](https://github.com/llvm/llvm-project/blob/main/mlir/tools/mlir-tblgen/EnumPythonBindingGen.cpp#L126). Will send a patch soon.

Reply:
Reverted for now

Reply:
Independent of Maks upcoming change, it's good to "textually namespace" TableGen side too. That's the common case, but I see unfortunately only documented such for ops and here you were being consistent with how fast math flags were defined. (This reminds me of the ODS linter again).